src/x.c

```diff
@@ -15,91 +15,148 @@ int16_t x_parseFile(
-
     corto_time start, stop;
     corto_route route = NULL;
     corto_id buffer;
     char *line;
     corto_int32 bytesRead = 0;
     corto_int32 i, lineCount = 0;
     corto_any param = {NULL}, result = {NULL, NULL};
-
     if (visitor) {
         param.type = corto_typeof(visitor);
         param.value = visitor;
     }
 
     printf("x/parser: parsing '%s'\n", file);
-
     corto_timeGet(&start);
     while ((line = corto_fileReadLine((corto_file)f, buffer, sizeof(buffer)))) {
         if (corto_router_match(instance, line, param, result, &route)) {
             corto_lasterr(); /* Suppress uncatched error warning */
         } else if (route) {
-            /* Keep track of how many times a route is called */ 
+            /* Keep track of how many times a route is called */
             for (i = 0; i < methods->length; i++) {
                 if (methods->buffer[i] == route) {
                     routesCalled.buffer[i] ++;
                 }
+
             }
+
             totalMatched ++;
         }
-        totalLines ++;
 
+        totalLines ++;
         bytesRead += strlen(line) + 1;
-
         if (!(lineCount % 5432)) {
             for (i = 0; i < 80; i++) {
                 printf("\b");
             }
-            printf("x/parser: %.1f%% (total = %d, matched = %d)", 
+
+            printf("x/parser: %.1f%% (total = %d, matched = %d)",
                 100.0 * ((float)bytesRead / (float)size), totalLines, totalMatched);
             fflush(stdout);
         }
+
         lineCount ++;
     }
+
     corto_timeGet(&stop);
     for (i = 0; i < 80; i++) {
         printf("\b");
     }
-    printf("x/parser: 100%% (total = %d, matched = %d)  \n", 
+
+    printf("x/parser: 100%% (total = %d, matched = %d)  \n",
         totalLines, totalMatched);
     printf("x/parser: done (%.1f seconds)\n",
         corto_timeToDouble(corto_timeSub(stop, start)));
-
     corto_trace("");
     corto_trace("x: Summary:");
     for (i = 0; i < routesCalled.length; i++) {
-        corto_trace("x: %30s: called %d times", 
-            corto_idof(methods->buffer[i]), 
+        corto_trace("x: %30s: called %d times",
+            corto_idof(methods->buffer[i]),
             routesCalled.buffer[i]);
     }
+
     corto_trace("");
-    corto_trace("x: parsed %d lines, matched %d in %fs", 
-        totalLines, 
+    corto_trace("x: parsed %d lines, matched %d in %fs",
+        totalLines,
         totalMatched,
         corto_timeToDouble(corto_timeSub(stop, start)));
-
     corto_dealloc(routesCalled.buffer);
 
+    fclose(f); /* Close file handler */
+
     return 0;
 error:
     return -1;
 }
 
+int16_t x_parseString(
+    corto_object instance,
+    corto_string line,
+    corto_object visitor)
+{
+    corto_routerimpl parser = corto_routerimpl(corto_typeof(instance));
+    corto_objectseq *methods = &corto_interface(parser)->methods;
+    corto_int32seq routesCalled;
+    routesCalled.buffer = corto_calloc(methods->length * sizeof(corto_int32));
```

Comment:
Not necessary (and not used). See below.

Reply:
@SanderMertens  Done